libs/repl/langchain_repl/repls/python.py

```diff
@@ -0,0 +1,71 @@
+import functools
+import logging
+import multiprocessing
+import sys
+from io import StringIO
+from typing import Dict, Optional
+
+from langchain.pydantic_v1 import BaseModel, Field
+
+logger = logging.getLogger(__name__)
+
+
+@functools.lru_cache(maxsize=None)
+def warn_once() -> None:
+    """Warn once about the dangers of PythonREPL."""
+    logger.warning("Python REPL can execute arbitrary code. Use with caution.")
```

Comment:
is default behavior not to warn once only under the `default` filter?

https://docs.python.org/3/library/warnings.html#the-warnings-filter